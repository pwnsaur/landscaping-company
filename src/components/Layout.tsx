import styled from 'styled-components';
import Footer from './Footer';
import Navigation from './Navigation';

type Props = {
  children: JSX.Element | JSX.Element[];
};

const Layout = ({ children }: Props) => {
  return (
    <Container>
      <Navigation />
      <Content>{children}</Content>
      <Footer />
    </Container>
  );
};

export default Layout;

const Container = styled.div`
  display: flex;
  flex-direction: column;
  min-height: 100vh;
  width: 100%;
  align-items: center;
  justify-content: center;
  background-color: ${({ theme }) => theme.colors.background};
`;

const Content = styled.div`
  display: flex;
  width: 100%;
  justify-content: center;
  /* margin-top: 2rem; */
  /* margin-bottom: 5rem; */
`;
