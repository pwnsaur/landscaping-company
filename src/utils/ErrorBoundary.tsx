import React, { ErrorInfo } from 'react';

interface Props {
  children: React.ReactNode;
}

interface State {
  hasError: boolean;
}

class ErrorBoundary extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);

    this.state = { hasError: false };
  }

  static getDerivedStateFromError(_: Error): State {
    return { hasError: true };
  }

  componentDidCatch(error: Error, errorInfo: ErrorInfo) {
    console.log({ error, errorInfo });
  }

  render() {
    if (this.state.hasError) {
      return (
        <div>
          <h1>Lapu neizdevās ielādēt.</h1>
        </div>
      );
    }

    return this.props.children;
  }
}

export default ErrorBoundary;
