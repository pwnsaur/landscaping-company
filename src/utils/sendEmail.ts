import axios, { AxiosResponse } from 'axios';

import { FormData } from '@/types/contactForm';

type ServerResponse = {
  messageId: string;
  response: string;
};

type SendEmailFn = (
  formData: FormData
) => Promise<AxiosResponse<ServerResponse>>;

const sendEmail: SendEmailFn = async (formData: FormData) => {
  return axios({
    method: 'post',
    url: '/api/send-mail',
    data: formData,
  });
};

export default sendEmail;
