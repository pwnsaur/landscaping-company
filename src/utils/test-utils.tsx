import { ReactElement, ReactNode } from 'react';
import { ThemeProvider } from 'styled-components';

import { theme } from '@/styles/theme';
import useIsMobile from '@/utils/hooks/useIsMobile';
import { render, RenderOptions } from '@testing-library/react';

interface TestWrapperProps {
  children: ReactNode;
}

const TestWrapper = ({ children }: TestWrapperProps) => {
  const isMobile = useIsMobile();

  return (
    <ThemeProvider theme={{ ...theme, isMobile }}>{children}</ThemeProvider>
  );
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'queries'>
) => {
  return render(ui, { wrapper: TestWrapper, ...options });
};

// eslint-disable-next-line import/export
export * from '@testing-library/react';

// eslint-disable-next-line import/export
export { customRender as render };
