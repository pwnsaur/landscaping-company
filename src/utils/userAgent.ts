export function isMobileUserAgent(userAgent: string): boolean {
  return /Android|iPhone|iPod|BlackBerry|IEMobile|Opera Mini/i.test(userAgent);
}
