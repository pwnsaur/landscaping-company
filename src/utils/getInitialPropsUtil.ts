import { AppContext, AppInitialProps } from 'next/app';

import { isMobileUserAgent } from './userAgent';

export async function getInitialPropsUtil(
  appContext: AppContext
): Promise<AppInitialProps & { isMobile: boolean }> {
  let pageProps = {};

  if (appContext.Component.getInitialProps) {
    pageProps = await appContext.Component.getInitialProps(appContext.ctx);
  }

  const userAgent = appContext.ctx.req?.headers['user-agent'] || 'rekt';
  const isMobile = isMobileUserAgent(userAgent);

  return { pageProps, isMobile };
}
