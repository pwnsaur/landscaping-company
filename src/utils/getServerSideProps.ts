import { GetServerSideProps } from 'next';

import { isMobileUserAgent } from '@utils/userAgent';

export const getServerSideProps: GetServerSideProps = async ({ req }) => {
  const userAgent = req.headers['user-agent'] || 'about rekt';
  const isMobile = isMobileUserAgent(userAgent);
  return { props: { isMobile } };
};
