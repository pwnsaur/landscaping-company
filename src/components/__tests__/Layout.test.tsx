import { render } from '@/utils/test-utils';
import Layout from '@/components/Layout';

jest.mock('next/router', () => ({
  useRouter: () => ({
    events: {
      on: jest.fn(),
      off: jest.fn(),
    },
  }),
}));

describe('MobileNav component', () => {
  test('matches the snapshot', () => {
    const { asFragment } = render(
      <Layout>
        <div>Would you test me? Id test me.</div>
      </Layout>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
