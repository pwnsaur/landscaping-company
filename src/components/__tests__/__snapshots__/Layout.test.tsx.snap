// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileNav component matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-75e6d97e-0 fWiFNT"
  >
    <div
      class="sc-ae6f8b53-0 cnJUxe"
    >
      <header
        class="sc-ae6f8b53-1 fNQlio"
      >
        <a
          class="sc-ae6f8b53-4 hZHTsL"
          href="/"
        >
          <img
            alt="logo"
            class="sc-ae6f8b53-3 jaFkLa"
            data-nimg="1"
            decoding="async"
            height="70"
            src="/_next/image?url=%2Fimg.jpg&w=256&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=128&q=75 1x, /_next/image?url=%2Fimg.jpg&w=256&q=75 2x"
            style="color: transparent;"
            width="120"
          />
        </a>
        <nav
          class="sc-23f60e78-0 gVZxNu"
        >
          <a
            class="sc-a1402bc8-0 NAwpH"
            href="/"
          >
            Sākums
          </a>
          <a
            class="sc-a1402bc8-0 NAwpH"
            href="/services"
          >
            Pakalpojumi
          </a>
          <a
            class="sc-a1402bc8-0 NAwpH"
            href="/projects"
          >
            Projekti
          </a>
          <a
            class="sc-a1402bc8-0 NAwpH"
            href="/about"
          >
            Par mums
          </a>
          <a
            class="sc-a1402bc8-0 NAwpH"
            href="/contacts"
          >
            Kontakti
          </a>
        </nav>
      </header>
    </div>
    <div
      class="sc-75e6d97e-1 kFlaaF"
    >
      <div>
        Would you test me? Id test me.
      </div>
    </div>
    <footer
      class="sc-d0132352-0 fRdPdt"
    >
      <a
        href="/contacts"
      >
        <button
          class="sc-990c1ae-0 hrFIOm"
        >
          Sazināties
        </button>
      </a>
      <p
        class="sc-d0132352-1 jYRPcv"
      >
        Copyright 2023 Hackerman
      </p>
    </footer>
  </div>
</DocumentFragment>
`;
